File: Backend/BCMarket_Construccion/Compra.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BCMarket_Construccion
{
    public class Compra
    {
        public int? Id_Compra { get; set; }
        public DateTime? Fecha { get; set; }
        public decimal? Total { get; set; }
        public string? Metodo_Entrega { get; set; }
        public int? Id_Transportista { get; set; }
        public string? Estado { get; set; }
        public List<DetalleCompra>? Detalles { get; set; }
        public Pago? Pago { get; set; }
        public string? Transaccion { get; set; }


    }
}

File: src/app/core/modelo/compra.ts
import { DetalleCompra } from "./detalleCompra";
import { Pago } from "./pago";

export interface Compra {
  id_Compra?: number;
  fecha?: string;
  total: number;
  metodo_Entrega: string;
  id_Transportista?: number;
  detalles: DetalleCompra[];
  pago: Pago;
  estado?: string;
  transaccion?: string;
}
